www/js/survey/enketo/enketo-add-note-button.js

```diff
@@ -0,0 +1,287 @@
+        }
+    }
+    // TODO: comment this out after testing to avoid log spew
+    console.log("Returning scrollElement ", $scope.scrollElement);
+    return $scope.scrollElement;
+  }
+
+  $scope.openPopover = function ($event, trip, inputType) {
+    const surveyName = $scope.notesConfig.surveyName;
+    console.log('About to launch survey ', surveyName);
+
+    let partialTimeUseResponse;
+    if (surveyName == 'TimeUseSurvey') {
+      partialTimeUseResponse = $scope.getPartialTimeUseResponse();
+    }
+
+    if ($event.stopPropagation) $event.stopPropagation();
+    return EnketoSurveyLaunch
+      .launch($scope, surveyName, { trip: trip, prefilledSurveyResponse: partialTimeUseResponse, dataKey: $scope.datakey })
+      .then(result => {
+        if (!result) {
+          return;
+        }
+        result.start_fmt_time = moment.unix(result.start_ts).format();
+        result.end_fmt_time = moment.unix(result.end_ts).format();
+        $scope.$apply(() => {
+          if (!trip.tripAddition)
+            trip.tripAddition = [];
+          trip.tripAddition.push({
+            data: result,
+            write_ts: Date.now()
+          });
+          const scrollElement = getScrollElement();
+          if (scrollElement) scrollElement.trigger('scroll-resize');
+        });
+        // store is commented out since the enketo survey launch currently
+        // stores the value as well
+        // $scope.store(inputType, result, false);
+      });
+  };
+})
+.factory("EnketoNotesButtonService", function(InputMatcher, EnketoSurveyAnswer, $timeout) {
+  var enbs = {};
+  console.log("Creating EnketoNotesButtonService");
+  enbs.SINGLE_KEY="NOTES";
+  enbs.key = "manual/trip_addition_input";
+  enbs.MANUAL_KEYS = [enbs.key];
+
+  /**
+   * Embed 'inputType' to the trip.
+   */
+   enbs.extractResult = (results) => EnketoSurveyAnswer.filterByNameAndVersion('TimeUseSurvey', results);
+
+   enbs.processManualInputs = function(manualResults, resultMap) {
+    if (manualResults.length > 1) {
+        Logger.displayError("Found "+manualResults.length+" results expected 1", manualResults);
+    } else {
+        console.log("ENKETO: processManualInputs with ", manualResults, " and ", resultMap);
+        const surveyResult = manualResults[0];
+        resultMap[enbs.SINGLE_KEY] = surveyResult;
+    }
+  }
+
+  enbs.populateInputsAndInferences = function(trip, manualResultMap) {
+    console.log("ENKETO: populating trip,", trip, " with result map", manualResultMap);
+    if (angular.isDefined(trip)) {
+        // console.log("Expectation: "+JSON.stringify(trip.expectation));
+        // console.log("Inferred labels from server: "+JSON.stringify(trip.inferred_labels));
+        if (!trip.trip_addition) {
+            trip.trip_addition = [];
+        }
+        trip.tripAddition = [];
+        enbs.populateManualInputs(trip, trip.nextTrip, enbs.SINGLE_KEY,
+            manualResultMap[enbs.SINGLE_KEY]);
```

Comment:
Note that a lot of this code can be simplified since there is only a SINGLE_KEY version of the trip additions - there is no MULTILABEL option. However, that is beyond the scope of this change.